### PR TITLE
 [#38] Add Hibernate with configuration to the project: again

### PR DESCRIPTION
**Pull request by GetFire**

### CODE REVIEW

File: src/main/java/net/greatstart/model/Project.java [hunk 1]
```diff
@@ -2,17 +2,37 @@
...
+    @Column(name = "min_invest")
     private int minInvestment;
-    private LocalDate addDate;
+
+    @ManyToOne(fetch = FetchType.EAGER)
     private User owner;
-    private List<User> investors;
+    /*
+     * Does not exist in DB table
+     * private List<User> investors;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/model/Project.java]: MAJOR This block of commented-out lines of code should be removed. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EventDaoImpl.java#L36](https://github.com/Sheptytskyid/Startup_Platform/blob/413726637ffee662595a187e923019b3e3ff964a/src/main/java/net/greatstart/dao/hibernate/EventDaoImpl.java#L36): Return an empty collection instead of null. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [InvestmentDaoImpl.java#L35](https://github.com/Sheptytskyid/Startup_Platform/blob/413726637ffee662595a187e923019b3e3ff964a/src/main/java/net/greatstart/dao/hibernate/InvestmentDaoImpl.java#L35): Return an empty collection instead of null. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [ProjectDaoImpl.java#L35](https://github.com/Sheptytskyid/Startup_Platform/blob/413726637ffee662595a187e923019b3e3ff964a/src/main/java/net/greatstart/dao/hibernate/ProjectDaoImpl.java#L35): Return an empty collection instead of null. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [UserDaoImpl.java#L40](https://github.com/Sheptytskyid/Startup_Platform/blob/413726637ffee662595a187e923019b3e3ff964a/src/main/java/net/greatstart/dao/hibernate/UserDaoImpl.java#L40): Return an empty collection instead of null. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: src/main/java/net/greatstart/dao/hibernate/UserDaoImpl.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import net.greatstart.dao.UserDao;
 import net.greatstart.model.User;
 import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dao/hibernate/UserDaoImpl.java]: MINOR Remove this unused import 'org.springframework.transaction.annotation.Transactional'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/net/greatstart/dao/hibernate/ProjectDaoImpl.java [hunk 3]
```diff
@@ -4,6 +4,7 @@
 import net.greatstart.model.Project;
 import org.springframework.stereotype.Repository;
 
+import javax.transaction.Transactional;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dao/hibernate/ProjectDaoImpl.java]: MINOR Remove this unused import 'javax.transaction.Transactional'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/net/greatstart/dao/hibernate/InvestmentDaoImpl.java [hunk 4]
```diff
@@ -3,6 +3,7 @@
 import net.greatstart.dao.InvestmentDao;
 import net.greatstart.model.Investment;
 import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dao/hibernate/InvestmentDaoImpl.java]: MINOR Remove this unused import 'org.springframework.transaction.annotation.Transactional'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/net/greatstart/dao/hibernate/EventDaoImpl.java [hunk 5]
```diff
@@ -3,6 +3,7 @@
 import net.greatstart.dao.EventDao;
 import net.greatstart.model.Event;
 import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dao/hibernate/EventDaoImpl.java]: MINOR Remove this unused import 'org.springframework.transaction.annotation.Transactional'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 8 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EventDaoImpl.java#L35](https://github.com/Sheptytskyid/Startup_Platform/blob/bc1d453bcd8feb5d2c013ad75ae03c2222808e6e/src/main/java/net/greatstart/dao/hibernate/EventDaoImpl.java#L35): Return an empty collection instead of null. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [InvestmentDaoImpl.java#L34](https://github.com/Sheptytskyid/Startup_Platform/blob/bc1d453bcd8feb5d2c013ad75ae03c2222808e6e/src/main/java/net/greatstart/dao/hibernate/InvestmentDaoImpl.java#L34): Return an empty collection instead of null. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [ProjectDaoImpl.java#L34](https://github.com/Sheptytskyid/Startup_Platform/blob/bc1d453bcd8feb5d2c013ad75ae03c2222808e6e/src/main/java/net/greatstart/dao/hibernate/ProjectDaoImpl.java#L34): Return an empty collection instead of null. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [UserDaoImpl.java#L39](https://github.com/Sheptytskyid/Startup_Platform/blob/bc1d453bcd8feb5d2c013ad75ae03c2222808e6e/src/main/java/net/greatstart/dao/hibernate/UserDaoImpl.java#L39): Return an empty collection instead of null. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Review by holkwen — APPROVED** [review]

**Pull request merged**